Add soil color data and change the way to read in Noah-MP parameter table

HelinWei-NOAA opened the pull request:
**Description**

- A new soil color dataset is introduced to Noah MP to reflect the right soil albedo to reduce the large warm bias found in the Sahel desert. The more-recent and higher resolution BNU soil texture data will be used to replace the old data. 

- Noah MP parameter table is part of the source code in the current version. The UFS model has to be recompiled whenever there is any parameter change in the table. Therefore we propose to change this structure and the table will become one of the input file to the model.

Closes #1614 
Closes #1617 


**Type of change**

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

<!-- Please describe the tests that you ran to verify your changes and on the platforms these tests were conducted. -->
<!-- Provide instructions so we can reproduce. -->
<!-- Please also list any relevant details for your test configuration -->

<!-- Use the following as a guide to list your tests and delete options that are not relevant. Expand as necessary. -->
<!--
- [ ] Clone and Build tests on WCOSS Dell P3
- [ ] Cycled test on Orion
- [ ] Forecast-only test on Hera
-->
  
**Checklist**

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

HelinWei-NOAA committed:
adding soil color data

WalterKolczynski-NOAA commented:
Since this PR does not include the NoahMP table stuff yet, I propose we make this PR just for the soil color data so we can push it through quickly, and then have a separate PR for the NoahMP table. (Single subject PRs are greatly preferred anyway.)

WalterKolczynski-NOAA commented:
@HelinWei-NOAA this has been sitting idle for two weeks. Is there more to add for the soil color data, or can we reduce the scope of this PR to get just that part in and use a separate PR for the Noah-MP parameter table?

HelinWei-NOAA commented:
@WalterKolczynski-NOAA I am working to get this into ufs repository first then I can start to work on global workflow

HelinWei-NOAA committed:
Merge remote-tracking branch 'upstream/develop' into hr2_land1

HelinWei-NOAA commented:
@WalterKolczynski-NOAA As you suggested I will split this PR to two. This one is just for soil color and the other one will be the NoahMP table. Can I open this PR for review?

WalterKolczynski-NOAA commented:
@HelinWei-NOAA Please fix the conflicts you now have with develop first, then open it up for review.

HelinWei-NOAA committed:
Merge pull request #2 from NOAA-EMC/develop

sync with the develop

HelinWei-NOAA commented:
@WalterKolczynski-NOAA I just synced my branch to the develop. What conflicts do you refer to? failing shellnorms/lint-shell check?

WalterKolczynski-NOAA commented:
Merging in develop fixed the conflicts. The shellcheck issues do need to be addressed though.

WalterKolczynski-NOAA commented:
This PR is undoing a lot of the changes in `forecast_postdet.sh` that we made to develop. Please review your changes and make sure only your intended changes are included.

HelinWei-NOAA committed:
merge with the latest develop

HelinWei-NOAA committed:
Merge branch 'hr2_land1' of github.com:HelinWei-NOAA/global-workflow into hr2_land1

HelinWei-NOAA commented:
@WalterKolczynski-NOAA Thanks for pointing this out.  I solved this issue but the commit seems too messy. Do you want me to start over with the develop branch? My modification is very minor.

> This PR is undoing a lot of the changes in `forecast_postdet.sh` that we made to develop. Please review your changes and make sure only your intended changes are included.

WalterKolczynski-NOAA commented:
It's fine as-is. We squash PRs into develop, so the history of the branch doesn't matter as long as the changes themselves are clean (which they appear to be now).

aerorahul reviewed (approved):
Just noting that this change won't be backward compatible due to the change in forecast_postdet.sh

Otherwise changes look good to me.

WalterKolczynski-NOAA commented:
> Just noting that this change won't be backward compatible due to the change in forecast_postdet.sh
> 
> Otherwise changes look good to me.

This was due to a change by me to remove `fix_` from the directory name in the new fix file set. You are correct though that the orog version can't be reverted without also undoing the postdet changes, but that is often true for fix file updates.

WalterKolczynski-NOAA commented:
@HelinWei-NOAA we had to revert a commit from earlier today that was causing issues with the CI. Please merge in develop again.

HelinWei-NOAA committed:
Merge branch 'NOAA-EMC:develop' into hr2_land1

WalterKolczynski-NOAA commented:
Does the UFS version need to be updated?
```
3: forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 12
```

HelinWei-NOAA commented:
@WalterKolczynski-NOAA The error was caused by missing the noahmp table. We need to get the other PR in before we can have a test. Or maybe we still should combine them as one PR.

HelinWei-NOAA commented:
@WalterKolczynski-NOAA I think we can't separate my changes into two PRs. Otherwise we can't do any successful test with the first one. So I will go ahead and merge the other one to here.

Commit:
changes for NoahMP table

Commit:
create parm/ufs/land

HelinWei-NOAA commented:
@WalterKolczynski-NOAA Yes. Now all of our changes have been merged to ufs-weather-model repos. You can start to test again. Thanks.
> Does the UFS version need to be updated?
> 
> ```
> 3: forrtl: severe (19): invalid reference to variable in NAMELIST input, unit -5, file Internal Formatted NML Read, line -1, position 12
> ```

WalterKolczynski-NOAA commented:
@HelinWei-NOAA You need to make the issues shellcheck found in your latest changes, then we can merge.

Commit:
meet ShellCheck

HelinWei-NOAA commented:
@WalterKolczynski-NOAA Done

HelinWei-NOAA commented:
@WalterKolczynski-NOAA The cause of those failed cases is the missing titled fixed files. For example C96C48_hybatmDA_b5607f97 on hera, the directory (/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/fix/orog/C96.mx500_frac) the script pointed to doesn't exist.

WalterKolczynski-NOAA commented:
> @WalterKolczynski-NOAA The cause of those failed cases is the missing titled fixed files. For example C96C48_hybatmDA_b5607f97 on hera, the directory (/scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/fix/orog/C96.mx500_frac) the script pointed to doesn't exist.

Yes it does:
```
hfe08 fcst.3189/ > ls -l /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/fix/orog/C96.mx500_frac
lrwxrwxrwx 1 role.glopara global 9 May 26 07:07 /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/fix/orog/C96.mx500_frac -> C96.mx500
hfe08 fcst.3189/ > ls -l /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/fix/orog/C96.mx500     
total 17M
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile1.nc
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile2.nc
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile3.nc
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile4.nc
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile5.nc
-rw-r--r-- 1 role.glopara stmp 814K May  5 15:30 C96.mx500_oro_data.tile6.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile1.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile2.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile3.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile4.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile5.nc
-rw-r--r-- 1 role.glopara stmp 2.0M May  5 15:30 C96_grid.tile6.nc
-rw-r--r-- 1 role.glopara stmp  22K May  5 15:30 C96_mosaic.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile1.nc -> C96.mx500_oro_data.tile1.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile2.nc -> C96.mx500_oro_data.tile2.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile3.nc -> C96.mx500_oro_data.tile3.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile4.nc -> C96.mx500_oro_data.tile4.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile5.nc -> C96.mx500_oro_data.tile5.nc
lrwxrwxrwx 1 role.glopara stmp   27 May  9 16:48 oro_C96.mx500.tile6.nc -> C96.mx500_oro_data.tile6.nc
```

I think the actual problem is there is no noahmptable.tbl file in the UFS you're checking out (there is no update to the UFS hash in this PR):
```
lrwxrwxrwx 1 Terry.McGuinness stmp    107 Jul 21 08:15 noahmptable.tbl -> /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/global-workflow/parm/ufs/land/noahmptable.tbl
```

Also, I see another thing: you are (trying to) link to the noahmptable in the UFS checkout, but you've _also_ added one to this repo. You should only be doing one of those (preferably linking once you have an updated UFS that has the file).

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -131,6 +131,10 @@ fi
 #---------------------------------------
+#--copy/link NoahMp table form ccpp-physics repository
+cd "${HOMEgfs}/parm/ufs/land" || exit 1
+  ${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl" .
```
Please remove the `noahmptable.tbl` from the global-workflow repository and link from the UFS-weather-model.

aerorahul commented on parm/ufs/land/noahmptable.tbl:
```diff
@@ -0,0 +1,780 @@
+&noahmp_usgs_veg_categories
```
Please remove this file from the global-workflow repository and use the one from the ufs-weather-model to link into the `global-workflow/parm/ufs/`. There is no need to create a `land` sub-directory under `parm/ufs`.

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -191,6 +191,9 @@ EOF
   fi
 
+# NoahMP table
+  ${NLN} "${HOMEgfs}/parm/ufs/land/noahmptable.tbl" "${DATA}/noahmptable.tbl" 
```
Please check first if this file is present, and if so, then link.
If it is not present, please abort with an appropriate message.

aerorahul commented on ush/forecast_postdet.sh:
```diff
@@ -304,6 +307,7 @@ EOF
   FNALBC=${FNALBC:-"${FIX_SFC}/${CASE}.snowfree_albedo.tileX.nc"}
   FNVETC=${FNVETC:-"${FIX_SFC}/${CASE}.vegetation_type.tileX.nc"}
   FNSOTC=${FNSOTC:-"${FIX_SFC}/${CASE}.soil_type.tileX.nc"}
+  FNSOCC=${FNSOCC:-"${FIX_SFC}/${CASE}.soil_color.tileX.nc"}
```
@WalterKolczynski-NOAA  Is this file available for all resolutions from `C48` all the way up to `C768`?

WalterKolczynski-NOAA replied:
From `C48` to `C3072`.

aerorahul reviewed (changes requested):
following the failed tests, I did a re-review of this PR.
Please see the comments and address accordingly.

Commit:
address comments from Rahul

WalterKolczynski-NOAA commented on sorc/link_workflow.sh:
```diff
@@ -131,10 +131,6 @@ fi
 #---------------------------------------
-#--copy/link NoahMp table form ccpp-physics repository
-cd "${HOMEgfs}/parm/ufs/land" || exit 1
-  ${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl" .
```
I think you misunderstood Rahul. This part was correct and is required by NCO. All the job script calls should look in this location, not down into `ufs_model.fd`. We just don't need the file in the repo since it will be linked in here (it should be added to the `.gitignore`)

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -192,7 +192,14 @@ EOF
 
 # NoahMP table
-  ${NLN} "${HOMEgfs}/parm/ufs/land/noahmptable.tbl" "${DATA}/noahmptable.tbl" 
+      local noahmptablefile="${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl"
```
See above comment. The location before was correct. The additional error checking is good though.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -192,7 +192,14 @@ EOF
-  ${NLN} "${HOMEgfs}/parm/ufs/land/noahmptable.tbl" "${DATA}/noahmptable.tbl" 
+      local noahmptablefile="${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl"
+    if [[ ! -f ${noahmptablefile} ]]; then
+      echo "ERROR: missing noahmp table file ${noahmptablefile}"
+      echo "Abort!"
```
```suggestion
      echo "FATAL ERROR: missing noahmp table file ${noahmptablefile}"
```

WalterKolczynski-NOAA reviewed (changes requested):
I still don't see a UFS model update. The current hash used by workflow ([bf60924](https://github.com/ufs-community/ufs-weather-model/tree/bf60924)) does not include the file `noahmptable.tbl`.

Commit:
modification based on Walter's comments

HelinWei-NOAA commented:
Sorry. I just updated the ufs hash.
> I still don't see a UFS model update. The current hash used by workflow ([bf60924](https://github.com/ufs-community/ufs-weather-model/tree/bf60924)) does not include the file `noahmptable.tbl`.

WalterKolczynski-NOAA commented on ush/forecast_postdet.sh:
```diff
@@ -191,6 +191,15 @@ EOF
     IEMS=${IEMS:-1}
   fi
 
+# NoahMP table
+      local noahmptablefile="${HOMEgfs}/parm/ufs/noahmptable.tbl"
+    if [[ ! -f ${noahmptablefile} ]]; then
+      echo "FATAL ERROR: missing noahmp table file ${noahmptablefile}"
+      exit 1
+    else
+      ${NLN} "${noahmptablefile}" "${DATA}/noahmptable.tbl"
+    fi
```
Please correct the indentation.

```suggestion
  # NoahMP table
  local noahmptablefile="${HOMEgfs}/parm/ufs/noahmptable.tbl"
  if [[ ! -f ${noahmptablefile} ]]; then
    echo "FATAL ERROR: missing noahmp table file ${noahmptablefile}"
    exit 1
  else
    ${NLN} "${noahmptablefile}" "${DATA}/noahmptable.tbl"
  fi
```

WalterKolczynski-NOAA reviewed (changes requested):
Please update the UFS hash in `Externals.cfg` as well.

HelinWei-NOAA committed:
Update ush/forecast_postdet.sh

Co-authored-by: Walter Kolczynski - NOAA <Walter.Kolczynski@noaa.gov>

HelinWei-NOAA committed:
Update Externals.cfg

WalterKolczynski-NOAA commented:
Failed build log: https://gist.github.com/WalterKolczynski-NOAA/e12e7829f95caeab76620d6d46d74f63

```
...

-- Compile stochastic_physics with 64-bit precision to match CCPP slow physics.
Calling CCPP code generator (ccpp_prebuild.py) for suites --suites=FV3_GFS_v16,FV3_GFS_v16_no_nsst,FV3_GFS_v16_ugwpv1,FV3_GFS_v17_p8,FV3_GFS_v16_coupled_nsstNoahmpUGWPv1,FV3_GFS_v17_coupled_p8 ...
CMake Error at FV3/ccpp/CMakeLists.txt:40 (message):
  An error occured while running ccpp_prebuild.py, check
  /work2/noaa/stmp/GFS_CI_ROOT/PR/1697/global-workflow/sorc/ufs_model.fd/build_fv3_0/FV3/ccpp/ccpp_prebuild.{out,err}
```

`ccpp_prebuild.err`:
```
INFO: Logging level set to INFO
INFO: Found TYPEDEFS_NEW_METADATA dictionary in config, assume at least some data is in new metadata format
INFO: Parsing suite definition files ...
INFO: Parsing suite definition file suites/suite_FV3_GFS_v16.xml ...
INFO: Parsing suite definition file suites/suite_FV3_GFS_v16_no_nsst.xml ...
CRITICAL: Suite definition file suites/suite_FV3_GFS_v16_no_nsst.xml not found.
ERROR: Parsing suite definition file suite_FV3_GFS_v16_no_nsst.xml failed.
Traceback (most recent call last):
  File "/work2/noaa/stmp/GFS_CI_ROOT/PR/1697/global-workflow/sorc/ufs_model.fd/FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 813, in <module>
    main()
  File "/work2/noaa/stmp/GFS_CI_ROOT/PR/1697/global-workflow/sorc/ufs_model.fd/FV3/ccpp/framework/scripts/ccpp_prebuild.py", line 723, in main
    raise Exception('Parsing suite definition files failed.')
Exception: Parsing suite definition files failed.
```

Looks like `FV3_GFS_v16_no_nsst` is no longer a valid CCPP suite. `sorc/build_ufs.sh` will need to be updated. I think we can get rid of all the v16 suites at this point.

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -131,6 +131,10 @@ fi
 #---------------------------------------
+#--copy/link NoahMp table form ccpp-physics repository
+cd "${HOMEgfs}/parm/ufs" || exit 1
+  ${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl" .
```
```suggestion
${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl" .
```

aerorahul reviewed (commented):
Just something minor.

aerorahul committed:
Update sorc/link_workflow.sh

aerorahul reviewed (approved):
Approve pending resolving updated model hash. 
This needs to be tested with the DA devs too to ensure DA applications will not fail after this model update.

HelinWei-NOAA commented:
@WalterKolczynski-NOAA Do you want to replace one with the general gfsv16 suite (suite_FV3_GFS_v16.xml)? So we will have it with another v17 suite (suite_FV3_GFS_v17_coupled_p8.xml) in build_ufs.sh

> 
> Looks like `FV3_GFS_v16_no_nsst` is no longer a valid CCPP suite. `sorc/build_ufs.sh` will need to be updated. I think we can get rid of all the v16 suites at this point.

WalterKolczynski-NOAA commented:
Nah, I don't think we need v16 suites at all anymore. Just keep the two v17 suites already in the suite list.

> @WalterKolczynski-NOAA Do you want to replace one with the general gfsv16 suite (suite_FV3_GFS_v16.xml)? So we will have it with another v17 suite (suite_FV3_GFS_v17_coupled_p8.xml) in build_ufs.sh
> 
> > Looks like `FV3_GFS_v16_no_nsst` is no longer a valid CCPP suite. `sorc/build_ufs.sh` will need to be updated. I think we can get rid of all the v16 suites at this point.

Commit:
update ufs_model hash and suite list

HelinWei-NOAA commented:
@barlage Do we have soil color data for C96.mx500?

HelinWei-NOAA commented:
> Automated global-workflow Testing Results:
> 
> ```
> Machine: Orion
> Start: Mon Jul 24 15:44:33 CDT 2023 on Orion-login-1.HPC.MsState.Edu
> ---------------------------------------------------
> Checkout:                      *SUCCESS*
> Checkout: Completed at Mon Jul 24 15:46:35 CDT 2023
> Build:                         *SUCCESS*
> Build: Completed at Mon Jul 24 16:14:43 CDT 2023
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 16:14:50 CDT 2023 for experiment C48_S2S_5c34fd9e
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 16:14:56 CDT 2023 for experiment C96_atm3DVar_5c34fd9e
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 16:15:01 CDT 2023 for experiment C96C48_hybatmDA_5c34fd9e
> Experiment C96_atm3DVar_5c34fd9e Terminated: *FAILED*
> Experiment C96_atm3DVar_5c34fd9e Terminated with 1 tasks failed at Mon Jul 24 16:26:12 CDT 2023
> Error logs:
> /work2/noaa/stmp/GFS_CI_ROOT/PR/1697/RUNTESTS/COMROT/C96_atm3DVar_5c34fd9e/logs/2021122018/gdasfcst.log
> ```
/work/noaa/global/glopara/fix/orog/20230615 doesn't exist

WalterKolczynski-NOAA commented:
> > Automated global-workflow Testing Results:
> > ```
> > Machine: Orion
> > Start: Mon Jul 24 15:44:33 CDT 2023 on Orion-login-1.HPC.MsState.Edu
> > ---------------------------------------------------
> > Checkout:                      *SUCCESS*
> > Checkout: Completed at Mon Jul 24 15:46:35 CDT 2023
> > Build:                         *SUCCESS*
> > Build: Completed at Mon Jul 24 16:14:43 CDT 2023
> > Created experiment:            *SUCCESS*
> > Case setup: Completed at Mon Jul 24 16:14:50 CDT 2023 for experiment C48_S2S_5c34fd9e
> > Created experiment:            *SUCCESS*
> > Case setup: Completed at Mon Jul 24 16:14:56 CDT 2023 for experiment C96_atm3DVar_5c34fd9e
> > Created experiment:            *SUCCESS*
> > Case setup: Completed at Mon Jul 24 16:15:01 CDT 2023 for experiment C96C48_hybatmDA_5c34fd9e
> > Experiment C96_atm3DVar_5c34fd9e Terminated: *FAILED*
> > Experiment C96_atm3DVar_5c34fd9e Terminated with 1 tasks failed at Mon Jul 24 16:26:12 CDT 2023
> > Error logs:
> > /work2/noaa/stmp/GFS_CI_ROOT/PR/1697/RUNTESTS/COMROT/C96_atm3DVar_5c34fd9e/logs/2021122018/gdasfcst.log
> > ```
> 
> /work/noaa/global/glopara/fix/orog/20230615 doesn't exist

That's my fault. I was having trouble transferring data and I guess I never synchronized the fix from Hera to Orion. It is running now.

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -131,6 +131,10 @@ fi
 #---------------------------------------
+#--copy/link NoahMp table form ccpp-physics repository
+cd "${HOMEgfs}/parm/ufs" || exit 1
+${LINK_OR_COPY} "${HOMEgfs}/sorc/ufs_model.fd/FV3/ccpp/physics/physics/noahmptable.tbl" .
```
Forgot to mention, but this file needs to be added to the list of ignored files in `.gitignore`

aerorahul reviewed (changes requested):
Needs an update in .gitignore

Commit:
update .gitignore

HelinWei-NOAA commented:
> Automated global-workflow Testing Results:
> 
> ```
> Machine: Hera
> Start: Mon Jul 24 20:45:18 UTC 2023 on hfe05
> ---------------------------------------------------
> Checkout:                      *SUCCESS*
> Checkout: Completed at Mon Jul 24 20:55:16 UTC 2023
> Build:                         *SUCCESS*
> Build: Completed at Mon Jul 24 21:25:00 UTC 2023
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 21:25:07 UTC 2023 for experiment C48_S2S_5c34fd9e
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 21:25:14 UTC 2023 for experiment C96C48_hybatmDA_5c34fd9e
> Created experiment:            *SUCCESS*
> Case setup: Completed at Mon Jul 24 21:25:19 UTC 2023 for experiment C96_atm3DVar_5c34fd9e
> Experiment C96C48_hybatmDA_5c34fd9e Terminated: *FAILED*
> Experiment C96C48_hybatmDA_5c34fd9e Terminated with 2 tasks failed at Mon Jul 24 21:42:30 UTC 2023
> Error logs:
> /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/RUNTESTS/COMROT/C96C48_hybatmDA_5c34fd9e/logs/2021122018/enkfgdasefcs01.log
> /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1697/RUNTESTS/COMROT/C96C48_hybatmDA_5c34fd9e/logs/2021122018/gdasfcst.log
> ```

The new soil color data is not provided for this resolution (C96.mx500). I just asked both Mike and Sanath if they did have them.

aerorahul commented:
Will it be added soon-ish?  We are in the process of adding a coupled cycling test at those resolutions?
This will impact coupled DA applications.

HelinWei-NOAA commented:
> Will it be added soon-ish? We are in the process of adding a coupled cycling test at those resolutions? This will impact coupled DA applications.

I found the data under /scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500. Just need Sanath confirm they are the right one @barlage @sanatcumar

barlage commented:
@HelinWei-NOAA can you tell me where the current test data are for C96.mx500?

WalterKolczynski-NOAA commented:
20230615 has now been properly sync'd to Orion. Will need that C96.mx500 to complete things though.

barlage commented:
I can confirm using this script: 

/home/Michael.Barlage/data/check/sfc_climo_check/check_land_soil_workflow.ncl

that the fix files in 

/scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500/C96.mx500.*

are consistent (valid soil type, veg type, soil color when land_frac>0) with the oro files in

/scratch1/NCEPDEV/global/glopara/fix/orog/20230615/C96.mx500

```
number of cumulative fix_oro_slmsk = 1                         : 18178
number of cumulative fix_oro_land_frac (0  ,1]                 : 18178
number of cumulative fix_oro_land_frac (0.5,1]                 : 16852
number of cumulative fix_oro_land_frac = 1                     : 15689
number of cumulative valid fix   veg type                      : 18178
number of cumulative valid fix  soil type                      : 18178
number of cumulative valid fix color type                      : 18178
number of cumulative missing fix   veg type with land_frac > 0 : 0
number of cumulative missing fix  soil type with land_frac > 0 : 0
number of cumulative missing fix color type with land_frac > 0 : 0
number of cumulative missing fix   veg type with     slmsk = 1 : 0
number of cumulative missing fix  soil type with     slmsk = 1 : 0
number of cumulative missing fix color type with     slmsk = 1 : 0
```

HelinWei-NOAA commented:
> 20230615 has now been properly sync'd to Orion. Will need that C96.mx500 to complete things though.

@WalterKolczynski-NOAA Please copy the data from /scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500
@barlage Thanks for the confirmation

WalterKolczynski-NOAA commented:
Fix files updated on Hera and Orion. Reset CI.

WalterKolczynski-NOAA commented:
```
15:  FATAL ERROR in sfcsub.F: No such file or directory
```

HelinWei-NOAA commented:
@WalterKolczynski-NOAA on both machines C96.mx500_frac is softlinked to the old C96.mx500 which doesn't include soil color data

WalterKolczynski-NOAA commented:
The files are there, but this new file set added an `.mx???` to the filename that wasn't there before. All of the soil fix file names will have to be updated for the new filename.

Old:
```
C96.facsf.tile1.nc                C96.slope_type.tile3.nc       C96.soil_color.tile5.nc             C96.vegetation_greenness.tile1.nc
C96.facsf.tile2.nc                C96.slope_type.tile4.nc       C96.soil_color.tile6.nc             C96.vegetation_greenness.tile2.nc
C96.facsf.tile3.nc                C96.slope_type.tile5.nc       C96.soil_type.tile1.nc              C96.vegetation_greenness.tile3.nc
C96.facsf.tile4.nc                C96.slope_type.tile6.nc       C96.soil_type.tile2.nc              C96.vegetation_greenness.tile4.nc
C96.facsf.tile5.nc                C96.snowfree_albedo.tile1.nc  C96.soil_type.tile3.nc              C96.vegetation_greenness.tile5.nc
C96.facsf.tile6.nc                C96.snowfree_albedo.tile2.nc  C96.soil_type.tile4.nc              C96.vegetation_greenness.tile6.nc
C96.maximum_snow_albedo.tile1.nc  C96.snowfree_albedo.tile3.nc  C96.soil_type.tile5.nc              C96.vegetation_type.tile1.nc
C96.maximum_snow_albedo.tile2.nc  C96.snowfree_albedo.tile4.nc  C96.soil_type.tile6.nc              C96.vegetation_type.tile2.nc
C96.maximum_snow_albedo.tile3.nc  C96.snowfree_albedo.tile5.nc  C96.substrate_temperature.tile1.nc  C96.vegetation_type.tile3.nc
C96.maximum_snow_albedo.tile4.nc  C96.snowfree_albedo.tile6.nc  C96.substrate_temperature.tile2.nc  C96.vegetation_type.tile4.nc
C96.maximum_snow_albedo.tile5.nc  C96.soil_color.tile1.nc       C96.substrate_temperature.tile3.nc  C96.vegetation_type.tile5.nc
C96.maximum_snow_albedo.tile6.nc  C96.soil_color.tile2.nc       C96.substrate_temperature.tile4.nc  C96.vegetation_type.tile6.nc
C96.slope_type.tile1.nc           C96.soil_color.tile3.nc       C96.substrate_temperature.tile5.nc
C96.slope_type.tile2.nc           C96.soil_color.tile4.nc       C96.substrate_temperature.tile6.nc

```

New:
```
C96.mx500.facsf.tile1.nc		C96.mx500.snowfree_albedo.tile1.nc  C96.mx500.substrate_temperature.tile1.nc
C96.mx500.facsf.tile2.nc		C96.mx500.snowfree_albedo.tile2.nc  C96.mx500.substrate_temperature.tile2.nc
C96.mx500.facsf.tile3.nc		C96.mx500.snowfree_albedo.tile3.nc  C96.mx500.substrate_temperature.tile3.nc
C96.mx500.facsf.tile4.nc		C96.mx500.snowfree_albedo.tile4.nc  C96.mx500.substrate_temperature.tile4.nc
C96.mx500.facsf.tile5.nc		C96.mx500.snowfree_albedo.tile5.nc  C96.mx500.substrate_temperature.tile5.nc
C96.mx500.facsf.tile6.nc		C96.mx500.snowfree_albedo.tile6.nc  C96.mx500.substrate_temperature.tile6.nc
C96.mx500.maximum_snow_albedo.tile1.nc	C96.mx500.soil_color.tile1.nc	    C96.mx500.vegetation_greenness.tile1.nc
C96.mx500.maximum_snow_albedo.tile2.nc	C96.mx500.soil_color.tile2.nc	    C96.mx500.vegetation_greenness.tile2.nc
C96.mx500.maximum_snow_albedo.tile3.nc	C96.mx500.soil_color.tile3.nc	    C96.mx500.vegetation_greenness.tile3.nc
C96.mx500.maximum_snow_albedo.tile4.nc	C96.mx500.soil_color.tile4.nc	    C96.mx500.vegetation_greenness.tile4.nc
C96.mx500.maximum_snow_albedo.tile5.nc	C96.mx500.soil_color.tile5.nc	    C96.mx500.vegetation_greenness.tile5.nc
C96.mx500.maximum_snow_albedo.tile6.nc	C96.mx500.soil_color.tile6.nc	    C96.mx500.vegetation_greenness.tile6.nc
C96.mx500.slope_type.tile1.nc		C96.mx500.soil_type.tile1.nc	    C96.mx500.vegetation_type.tile1.nc
C96.mx500.slope_type.tile2.nc		C96.mx500.soil_type.tile2.nc	    C96.mx500.vegetation_type.tile2.nc
C96.mx500.slope_type.tile3.nc		C96.mx500.soil_type.tile3.nc	    C96.mx500.vegetation_type.tile3.nc
C96.mx500.slope_type.tile4.nc		C96.mx500.soil_type.tile4.nc	    C96.mx500.vegetation_type.tile4.nc
C96.mx500.slope_type.tile5.nc		C96.mx500.soil_type.tile5.nc	    C96.mx500.vegetation_type.tile5.nc
C96.mx500.slope_type.tile6.nc		C96.mx500.soil_type.tile6.nc	    C96.mx500.vegetation_type.tile6.nc
```

HelinWei-NOAA commented:
@WalterKolczynski-NOAA I see. You can remove .mx500 from the file name FYI@barlage

WalterKolczynski-NOAA commented:
The entire new fix set is like this. The other (non-sfc) files also have the mx, so I'd rather just update the filenames in the script (it should be easier too).

sanatcumar commented:
Hi Walter,
 /scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500/C96.mx500
has the surface field with and without the "mx500".  If not could you point
me to the directory  it was pulled form please ?
Cheers
Sanath


On Wed, Jul 26, 2023 at 12:45 PM Walter Kolczynski - NOAA <
***@***.***> wrote:

> The entire new fix set is like this. The other (non-sfc) files also have
> the mx, so I'd rather just update the filenames in the script (it should be
> easier too).
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/NOAA-EMC/global-workflow/pull/1697#issuecomment-1652166305>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AZGCV3M4B7Y4CAUQBNB277TXSFCTJANCNFSM6AAAAAAZIGZ4AA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sanatcumar commented:
Hi Walter,
 To avoid confusion, I have removed the filenames with the "mx500" in my
directory "/scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500"
Cheers
Sanath





On Wed, Jul 26, 2023 at 2:34 PM Sanath Kumar - NOAA Affiliate <
***@***.***> wrote:

>  Hi Walter,
>  /scratch2/NCEPDEV/stmp1/Sanath.Kumar/sfc.C96.mx500/C96.mx500
> has the surface field with and without the "mx500".  If not could you
> point me to the directory  it was pulled form please ?
> Cheers
> Sanath
>
>
> On Wed, Jul 26, 2023 at 12:45 PM Walter Kolczynski - NOAA <
> ***@***.***> wrote:
>
>> The entire new fix set is like this. The other (non-sfc) files also have
>> the mx, so I'd rather just update the filenames in the script (it should be
>> easier too).
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/NOAA-EMC/global-workflow/pull/1697#issuecomment-1652166305>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AZGCV3M4B7Y4CAUQBNB277TXSFCTJANCNFSM6AAAAAAZIGZ4AA>
>> .
>> You are receiving this because you were mentioned.Message ID:
>> ***@***.***>
>>
>

WalterKolczynski-NOAA committed:
Add mask resolution to sfc fix file names

The newest version of the surface fix files includes the mask resolution
(`mx.###`) in the file name. This brings them in line with other non-sfc
fix files that already include the mask in their filename.

Refs #1697

WalterKolczynski-NOAA commented:
@HelinWei-NOAA I opened a PR to the branch in your fork yesterday that should update all the filenames (I didn't test it). If you would, please review/merge that and then run a test using a clean copy before we throw it at the CI testing again.

HelinWei-NOAA committed:
Merge pull request #4 from WalterKolczynski-NOAA/hr2_land1

Add mask resolution to sfc fix file names

HelinWei-NOAA commented:
@WalterKolczynski-NOAA Merged and ran a successful prototype test.

aerorahul reviewed (approved):
Looks good. 
Thanks for the work @HelinWei-NOAA and team.
